src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -711,4 +712,65 @@ BOOST_AUTO_TEST_CASE(MarginRankingLossTest)
       "-0.753830 1.336900 0.000000 0.000000 -0.207000 0.328810"), 1e-6);
 }
 
+/**
+ * Simple test for the multiLabel softmargin loss function.
+ */
+BOOST_AUTO_TEST_CASE(MultiLabelSoftMarginLossSumReductionTest)
+{
+  arma::mat input, target, output, weight, expectedOutput;
+  double loss;
+  weight.ones(1, 3);
+  MultiLabelSoftMarginLoss<> module(weight, 3, true);
+
+  input = arma::mat("0.1778 0.1203 -0.2264 0.0957 0.2403 -0.3400 0.1397 0.1925 "
+      "-0.3336");
+  target = arma::mat("0 1 0 1 0 0 0 0 1");
+  expectedOutput = arma::mat("0.1814 -0.1567 0.1479 -0.1587 0.1866 0.1386 "
+      "0.1783 0.1827 -0.1942");
+  input.reshape(3, 3);
+  target.reshape(3, 3);
+  expectedOutput.reshape(3, 3);
+
+  // Test the Forward function. Loss should be 2.14829.
+  loss = module.Forward(input, target);
+  BOOST_REQUIRE_CLOSE(loss, 2.14829, 1e-3);
+
+  // Test the Backward function.
+  module.Backward(input, target, output);
+  BOOST_REQUIRE_CLOSE(arma::as_scalar(arma::accu(output)), 0.505909, 1e-3);
+  BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
+  BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
+  CheckMatrices(output, expectedOutput, 0.1);
+}
+
+/**
+ * Simple test for the multiLabel softmargin loss function.
+ */
+BOOST_AUTO_TEST_CASE(MultiLabelSoftMarginLossMeanReductionTest)
+{
+  arma::mat input, target, output, weight, expectedOutput;
+  double loss;
+  weight.ones(1, 3);
```

Comment:
The reason I felt weight matrix as input is redundant because of this. User shouldn't set this to 1 if he doesn't want to change weight for different classes.

Reply:
I have thought of this too. The reason why I keep wanting to go with the current implementation is simply that I feel users should be consciously choosing the weights when they are using a very specific loss function such as this one. So, even if they want to go with the default implementations, they initialise a weight vector and also if they want to give different weights to each class, they have to initialise a weight vector. That's the idea I am going for. I totally get your point. But, I would like at least one more person's opinion on this before I  change the implementation. Let me know if you have further suggestions. Thanks!